File: WPServer/WPProjectServer/WPProjectServer/Logic/SubsystemLogic/UserSubLogic.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using WPProjectServer.DatabaseObjects;

namespace WPProjectServer.Logic.SubsystemLogic
{
    public class UserSubLogic : DataPullSubLogic
    {
        public bool ChangeUserPenalty(User user, decimal penaltyToAdd)
        {
            user.Debt += penaltyToAdd;
            return SaveUser(user);
        }

        public bool ChangeUserPenalty(string userCode, decimal penaltyToAdd)
        {
            User user = GetUser(userCode);
            if (user == null)
                throw new Exception("Brak uzytkownika o danym kodzie!");

            user.Debt += penaltyToAdd;
            return SaveUser(user);
        }

        public bool SaveUser(User user)
        {
            return true;
        }

        public bool RegisterUser(User user)
        {
            return true;
        }
    }
}

File: WPWeb/Library/src/main/java/controllers/SubLogic/UserSubLogic.java
package controllers.SubLogic;

import controllers.LogicFactory;
import models.User;

/**
 * Created by Szymon on 2015-10-13.
 */

public class UserSubLogic extends DataPullSubLogic
{
    public UserSubLogic(LogicFactory lgF)
    {
        super(lgF);
    }

    public boolean ChangeUserPenalty(User user, long penaltyToAdd)
    {
        user.debt += penaltyToAdd;
        return SaveUser(user);
    }

    public boolean ChangeUserPenalty(String userCode, double penaltyToAdd) throws Exception {
        User user = GetUser(userCode);
        if (user == null)
            throw new Exception("Brak uzytkownika o danym kodzie!");

        user.debt += penaltyToAdd;
        return SaveUser(user);
    }

    public boolean SaveUser(User user)
    {
        return true;
    }

    public boolean RegisterUser(User user)
    {
        return true;
    }
}
